spec/models/dataset_file_spec.rb

```diff
@@ -12,35 +12,60 @@
       hash = {
-        name: "10.34770/qyrs-vg25/50/file_name.txt",
-        full_name: "file_name.txt",
+        filename: "10.34770/qyrs-vg25/50/file_name.txt",
+        name: "file_name.txt",
         size: 455_511,
         url: "https://g-ef94ef.f0ad1.36fe.data.globus.org/10.34770/qyrs-vg25/50/file_name.txt"
       }
       file = described_class.from_hash(hash, "pdc_describe")
       expect(file.download_url).to eq hash[:url]
     end
   end
+
+  describe "#display_filename" do
+    it "handles normal cases correctly" do
+      expect(described_class.filename_without_doi("10.123/4567/40/file1.txt")).to eq "file1.txt"
+      expect(described_class.filename_without_doi("10.123/4567/40/folder1/file1.txt")).to eq "folder1/file1.txt"
+    end
+
+    it "defaults to the full path when DOI is not present" do
+      expect(described_class.filename_without_doi("file1.txt")).to eq "file1.txt"
+      expect(described_class.filename_without_doi("folder1/file1.txt")).to eq "folder1/file1.txt"
+    end
+
+    it "defaults to full path when database ID is not numeric" do
+      expect(described_class.filename_without_doi("10.123/4567/x40/file1.txt")).to eq "10.123/4567/x40/file1.txt"
+    end
+
+    it "handles nulls and empty values" do
+      expect(described_class.filename_without_doi("")).to eq ""
+      expect(described_class.filename_without_doi(nil)).to eq ""
+    end
+  end
+
   describe "sort Dataset files by name and size" do
     let(:file1) do
       DatasetFile.from_hash({
                               name: "b.txt",
+                              filename: "10.34770/bm4s-t361/89/b.txt",
                               size: "455511",
-                              download_url: "https://g-ef94ef.f0ad1.36fe.data.globus.org/10.34770/qyrs-vg25/50/file_name.txt"
+                              download_url: "https://g-ef94ef.f0ad1.36fe.data.globus.org/10.34770/bm4s-t361/89/b.txt"
                             }, "pdc_describe")
     end
     let(:file2) do
       DatasetFile.from_hash({
                               name: "a.txt",
+                              filename: "10.34770/bm4s-t361/89/a.txt",
                               size: "19271048",
-                              download_url: "https://g-ef94ef.f0ad1.36fe.data.globus.org/10.34770/bm4s-t361/89/Fig9b.hdf"
+                              download_url: "https://g-ef94ef.f0ad1.36fe.data.globus.org/10.34770/bm4s-t361/89/a.txt"
                             }, "pdc_describe")
     end
     let(:file3) do
       DatasetFile.from_hash({
                               name: "README.txt",
+                              filename: "10.34770/bm4s-t361/89/README.txt",
                               size: "5173",
-                              download_url: "https://g-ef94ef.f0ad1.36fe.data.globus.org/10.34770/bm4s-t361/89/readme.txt"
+                              download_url: "https://g-ef94ef.f0ad1.36fe.data.globus.org/10.34770/bm4s-t361/89/README.txt"
```

Comment:
Our new code relies on some of the keys in the hash that are always present in the data but that were no present in our test data or were populated inconsistently (like the `download_url` didn't match the actual file `name` in the hash)